### DISCUSSION TITLE
How to produce animation with Warp

### DISCUSSION THREAD

**Discussion by yuhan-zh** [NVIDIA/warp#165] [Q&A]
Hello!
       I am wondering how can I produce animation with Warp like what is shown in the first gif (human with dress dancing) in the Readme.
       More specifically, I already had a method that can produce a sequence of human body mesh (which produce a smooth body movement). How can I use this sequence of mesh in the simulation to produce an animation? As I understand, Warp considers bodies/shapes as rigid and only allows a transformation on them.  If the Warp doesn't support this, do you have any recommendations for me to implement it?

Thanks a lot!

**Comment by mmacklin** [comment]
Hi @yuhan-zh, Warp supports collision with deforming bodies / meshes, please take a look at https://github.com/NVIDIA/warp/blob/main/examples/example_mesh.py#L126C19-L126C19 for an example of how to update an existing `wp.Mesh` object. 

For the dancing character we exported time-sampled USD vertex animation from Blender, then per-frame we update the collision geometry accordingly. Please take a look at https://github.com/NVIDIA/warp/blob/main/exts/omni.warp/omni/warp/nodes/OgnCloth.py for an example of how the Omniverse integration looks like. This also shows how to compute per-vertex velocities given the mesh at two keyframes.

Hope that helps.

Best,
Miles

**Comment by yuhan-zh** [comment]
@mmacklin Thanks for your reply! The recommendations help a lot. And btw I really hope you can release the example code of the dancing character, it really looks amazing.

Best,
Yuhan
